File: models/bloodbank/BloodDiscard.js
// models/BloodDiscard.js
import mongoose from "mongoose";

const BloodDiscardSchema = new mongoose.Schema(
  {
    hospitalId: { type: mongoose.Schema.Types.ObjectId, ref: "Hospital", required: true },
    branchId: { type: mongoose.Schema.Types.ObjectId, ref: "Branch", required: true },

    componentId: { type: mongoose.Schema.Types.ObjectId, ref: "BloodComponent", required: true },

    reason: String,
    discardedBy: { type: mongoose.Schema.Types.ObjectId, ref: "User" },
    discardDate: Date,
  },
  { timestamps: true }
);

export default mongoose.model("BloodDiscard", BloodDiscardSchema);

File: jstojv/src/main/java/com/samrat/jstojv/entity/BloodDiscard.java
package com.samrat.jstojv.entity;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "blood_discards")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class BloodDiscard extends BaseEntity {

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "hospital_id", nullable = false)
    private Hospital hospital;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "branch_id", nullable = false)
    private Branch branch;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "component_id", nullable = false)
    private BloodComponent component;

    private String reason;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "discarded_by")
    private User discardedBy;

    @Column(name = "discard_date")
    private LocalDateTime discardDate;
}
